### PR TITLE
Feat: Adjust Newsletter to match new designs

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

Updates the `Newsletter` component implementation according to the prototype in Figma. (Some updates were made while developing it)

## How does it work?

- Updates `Newsletter` component style (adds `card` and `lite` variations) [Preview](https://nextjs-store-storybook-git-feat-fs-4941-adjust-c017a7-faststore.vercel.app/?path=/docs/organisms-newsletter--default-story)

|Default - Deskstop| Default - Mobile|
|-|-|
image|image|

- Uses `Toast` to display Newsletter subscription feedback
- Adds `Newsletter` to Homepage and PDP
- Adds `Privacy Policy` addendum message to `Newsletter`
- Adds Toast Tokens documentation ([Preview](https://nextjs-store-storybook-git-feat-fs-4941-adjust-c017a7-faststore.vercel.app/?path=/docs/atoms-toast--default-story))

## How to test it?

- Check if `Newsletter` is being displayed on [Homepage](https://sfj-6dc3661--nextjs.preview.vtex.app/)

| Homepage - Mobile| Homepage - Desktop|
|-|-|
|image|image|
 
- Check if `Toast` is being displayed correctly when subscribing 

|Error| Success|
|-|-|
|image|image|

- Check if `BannerNewsletter` is being displayed on [PDP page ](https://sfj-6dc3661--nextjs.preview.vtex.app/4k-philips-monitor-99988213/p)

|PDP - Mobile| PDP - Desktop|
|-|-|
|image|image|

## References

[FS-494](https://vtex-dev.atlassian.net/browse/FS-494?atlOrigin=eyJpIjoiODQyNmQ1Mjg1MDUwNGVlM2I4ZWRhNTc3YWMxY2I2MTEiLCJwIjoiaiJ9)

## Checklist

<em>You may erase this after checking them all :wink:</em>

**PR Title and Commit Messages**
- [x] PR title and commit messages follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification
  - Available prefixes: `feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, and `test`

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Features`, `Bug fixes`, `Chore`, `Documentation`, `Style changes`, `Refactoring`, `Performance`, and `Test`
- [x] Added the component, hook, or path name in-between backticks (\`\`) - *if applicable, e.g., `ComponentName` component, `useWindowDimensions` hook*

**Dependencies**
- [x] Committed the `yarn.lock` and `bun.lockb` file when there were changes to the packages

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*
- [ ] For documentation changes, ping `@carolinamenezes` or `@PedroAntunesCosta` to review and update

### CODE REVIEW

File: src/components/ui/Toast/Toast.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+<TokenTable>
+  <TokenRow
+    token="--fs-toast-message-size"
+    value="var(--fs-toast-title-size)"
+  />
+  <TokenRow
+    token="--fs-toast-message-size"
+    value="var(--fs-toast-title-line-height)"
+  />
+  <TokenRow token="--fs-toast-message-size" value="var(--fs-spacing-3)" />
```
Comment (renatamottam) [inline src/components/ui/Toast/Toast.stories.mdx:158]:
```suggestion
  <TokenRow
    token="--fs-toast-message-line-height"
    value="var(--fs-toast-title-line-height)"
  />
  <TokenRow token="--fs-toast-message-margin-left" value="var(--fs-spacing-3)" />
```

File: src/components/sections/BannerNewsletter/banner-newsletter.module.scss [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+.fs-banner-newsletter {
+  margin-bottom: var(--fs-grid-padding);
+
+  > section {
+    margin-top: 0;
+  }
+
+  @include media(">=notebook") {
+    display: flex;
```
Comment (renatamottam) [inline src/components/sections/BannerNewsletter/banner-newsletter.module.scss:11]:
```suggestion
    display: flex;
    column-gap: var(--fs-grid-gap-3);
```

File: src/components/sections/Newsletter/newsletter.module.scss [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    [data-fs-button-inverse] {
+      margin-top: var(--fs-spacing-3);
+    }
+  }
+
+  // --------------------------------------------------------
+  // Variants Styles
+  // --------------------------------------------------------
+
+  &[data-fs-newsletter="card"] {
```
Comment (renatamottam) [inline src/components/sections/Newsletter/newsletter.module.scss:105]: Fan, the texts should be center aligned on `Card` variant 🙃

File: src/components/sections/BannerNewsletter/BannerNewsletter.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import BannerText from '../BannerText'
+import Newsletter from '../Newsletter'
+import Section from '../Section'
+import styles from './banner-newsletter.module.scss'
+
+function BannerNewsletter() {
+  return (
+    <Section data-fs-product-details-content className="layout__content">
+      <div className={styles.fsBannerNewsletter}>
+        <BannerText
```
Comment (renatamottam) [inline src/components/sections/BannerNewsletter/BannerNewsletter.tsx:10]: Ideally, this guy should also use `var(--fs-border-radius)`

**Review by renatamottam — COMMENTED** [review]
Nice work, fan! Left a few improvement suggestions :)

File: src/components/sections/Newsletter/newsletter.module.scss (hunk 3 above)
Comment (hellofanny) [reply src/components/sections/Newsletter/newsletter.module.scss:105]: Okay! Lemme fix It :)

File: src/components/sections/BannerNewsletter/BannerNewsletter.tsx (hunk 4 above)
Comment (hellofanny) [reply src/components/sections/BannerNewsletter/BannerNewsletter.tsx:10]: I've [added](https://github.com/vtex-sites/nextjs.store/pull/252/commits/319c1971c5ac8a45113e8e831594f4cb87825420) to `BannerText`. Is that correct?

**Review by renatamottam — APPROVED** [review]

File: src/components/sections/BannerNewsletter/BannerNewsletter.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import BannerText from '../BannerText'
+import Newsletter from '../Newsletter'
+import Section from '../Section'
+import styles from './banner-newsletter.module.scss'
+
+function BannerNewsletter() {
+  return (
+    <Section data-fs-product-details-content className="layout__content">
```
Comment (lucasfp13) [inline src/components/sections/BannerNewsletter/BannerNewsletter.tsx:8]:
```suggestion
    <Section className="layout__content">
```

**Review by lucasfp13 — APPROVED** [review]

File: src/components/sections/BannerNewsletter/BannerNewsletter.tsx (hunk 5 above)
Comment (hellofanny) [reply src/components/sections/BannerNewsletter/BannerNewsletter.tsx:8]: opps my bad.

**Pull request merged**